src/types.ts

```diff
@@ -1,26 +1,90 @@
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+import { ethers } from 'ethers';
+
+import { SupportedChainId } from './config';
 import { Asset, Transaction } from './gnosis-safe/types';
-import { Proposal, TokenMetaData, Vote, zDAO } from './snapshot-io/types';
+import {
+  Proposal,
+  ProposalDetail,
+  ProposalResult,
+  TokenMetaData,
+  Vote,
+} from './snapshot-io/types';
 
 export interface SnapshotConfig {
+  // uri to Snaphost Hub
   serviceUri: string;
-  ipfsUri: string;
+  // ipfs gateway
+  ipfsGateway: string;
 }
 
 export interface GnosisSafeConfig {
+  // uri to Gnosis Safe service
   serviceUri: string;
-  safeAddress: string;
+  // uri to Gnosis Safe gateway
   gateway: string;
 }
 
+export interface zNAConfig {
+  // address to zDAOCore contract
+  contract: string;
+  // web3 provider
+  provider: ethers.providers.Provider;
+}
+
 export interface Config {
   snapshot: SnapshotConfig;
   gnosisSafe: GnosisSafeConfig;
-  chainId: string;
+  zNA: zNAConfig;
+}
+
+export type zNA = string;
+export type zDAOId = string;
+
+export interface zDAO {
+  id: zDAOId; // Global zDAO identifier
+  zNA: zNA; // Linked zNA
+  // If not defined, title should be zNA by default
+  title?: string; // zDAO title
+  creator: string; // Creator wallet address
+  owners: string[]; // Owner wallet addresses
+  // avatar uri which starts with https schema
+  // The frontend should use default avatar image if not defined
+  avatar?: string; // Avatar uri (https link)
+  network: string; // Chain id
+  safeAddress: string; // Gnosis Safe address
+  votingToken: string; // Voting token address
+}
+
+export interface CreateZDAODto {
+  // zNA
+  zNA: zNA;
+  // zDAO title
+  title: string;
+  // address to zDAO creator
+  creator: string;
+  // uri to avatar, if not defined, will use default avatar image in frontend
+  avatar?: string;
+  // network id where zDAO was created
+  network: SupportedChainId;
+  // adress to Gnosis Safe
+  safeAddress: string;
+  // addresses to Gnosis Safe owners
+  owners: string[];
+  // ERC20 token address to cast a vote
+  votingToken: string;
+}
+
+export interface zDAOAssets {
+  // total asset amount in USD
+  amountInUSD: number;
+  // list of assets in zDAO
+  assets: Asset[];
```

Comment:
Instead of importing `Asset` define a different `Asset` interface in this file.

The reason being is that we may remove the gnosis safe dependency and the `Asset` structure would then change.

This would also mean that the code in `gnosis-safe/` would need to return this type of `Asset` and not it's own